lib/rules/no-unsafe-finally.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @fileoverview Rule to flag unsafe statements in finally block
+ * @author Onur Temizkan
+ * @copyright 2016 Onur Temizkan. All rights reserved.
+ * See LICENSE in root directory for full license.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var Traverser = require("../util/traverser");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+var UNSAFE_STATEMENTS = [
+    "ReturnStatement",
+    "ThrowStatement",
+    "BreakStatement",
+    "ContinueStatement"
+];
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "Flags unsafe statements in finally block",
+            category: "Possible Errors",
+            recommended: false
+        }
+    },
+    create: function(context) {
+
+        var traverser = new Traverser();
+
+        /**
+         * Checks if node has an unsafe statement in finally.
+         *
+         * @param {ASTNode} node - node to check.
+         * @returns {void}
+         */
+        function check(node) {
+            if (!node.finalizer) {
+                return;
+            }
+
+            traverser.traverse(node.finalizer, {
```

Comment:
My bad. Since @onurtemizkan and I are friends I carried on the conversation on a private channel instead of writing here. Will do it public next time so everyone can benefit and chime in. :)